library/ssl_tls13_generic.c

```diff
@@ -1199,6 +1200,19 @@ int mbedtls_ssl_reset_transcript_for_hrr( mbedtls_ssl_context *ssl )
 #if defined(MBEDTLS_SHA256_C) || defined(MBEDTLS_SHA384_C)
     ssl->handshake->update_checksum( ssl, hash_transcript, hash_len );
 #endif /* MBEDTLS_SHA256_C || MBEDTLS_SHA384_C */
+
+    /* Destroy generated private key. */
+    status = psa_destroy_key( ssl->handshake->ecdh_psa_privkey );
```

Comment:
@mprse @mpg @xffbai I think this should be in 'ssl_tls13_reset_key_share` which was not changed by this PR and still call  `mbedtls_ecdh_free()`. Do I miss something ?